python/src/equistore/tensor.py

```diff
@@ -68,6 +78,23 @@ def _from_ptr(ptr):
         obj._blocks = []
         return obj
 
+    @classmethod
+    def _from_pickle(cls, tm_bytes: bytes):
+        """
+        Passed to pickler to reconstruct TensorMap from bytes object
+        """
+        lib = _get_library()
+        path = os.path.join(tempfile.gettempdir(), str(id(tm_bytes)))
+        try:
+            with open(path, "wb") as f:
+                pickle.dump(tm_bytes, f)
+            ptr = lib.eqs_tensormap_load(
+                path.encode("utf8"), eqs_create_array_callback_t(create_numpy_array)
+            )
```

Comment:
Why not `tensor = equistiore.io.load(...)` here?

Reply:
I think this was because io.py has a dependency on TensorMap, so I could not import it here

Reply:
right, you can not import it on the top level, but I think importing inside the function should be fine. We do something similar for `__eq__` and friends

Reply:
And what you think about the `create_numpy_array`  function? I don't need it anymore in the tensor.py for this PR, so I can remove the dependency and also move it back to io.py. But for the step 1, I might need to pass create_numpy_array 
```python
    @classmethod
    def _from_pickle(cls, data: bytes):
        """
        Passed to pickler to reconstruct TensorMap from bytes object
        """
        lib = _get_library()
        return lib.eqs_tensormap_load_from_buffer(data, eqs_create_array_callback_t(create_numpy_array))
```
except if we keep it in the io.py as well
```python
    @classmethod
    def _from_pickle(cls, data: bytes):
        """
        Passed to pickler to reconstruct TensorMap from bytes object
        """
        from equistore.io import load_from_buffer
        return load_from_buffer(data)
```
What do you prefer?

Reply:
I would leave everything related to io in the `io` module